packages/valory/skills/abstract_round_abci/tests/test_tools/test_rounds.py

```diff
@@ -0,0 +1,689 @@
+
+
+settings(deadline=5000)
+
+
+def test_get_participants() -> None:
+    """Test `get_participants`."""
+    participants = get_participants()
+    assert isinstance(participants, frozenset)
+    assert all(isinstance(p, str) for p in participants)
+    assert len(participants) == MAX_PARTICIPANTS
+
+
+class DummyTxPayloadMatcher:
+    """A `DummyTxPayload` matcher for assertion comparisons."""
+
+    expected: DummyTxPayload
+
+    def __init__(self, expected: DummyTxPayload) -> None:
+        """Initialize the matcher."""
+        self.expected = expected
+
+    def __repr__(self) -> str:
+        """Needs to be implemented for better assertion messages."""
+        return (
+            "DummyTxPayload("
+            f"id={repr(self.expected.id_)}, "
+            f"round_count={repr(self.expected._round_count)}, "
+            f"sender={repr(self.expected.sender)}, "
+            f"value={repr(self.expected._value)}, "
+            f"vote={repr(self.expected._vote)}"
+            ")"
+        )
+
+    def __eq__(self, other: Any) -> bool:
+        """The method that will be used for the assertion comparisons."""
+        return (
+            self.expected._round_count == other._round_count
+            and self.expected.sender == other.sender
+            and self.expected._value == other._value
+            and self.expected._vote == other._vote
+        )
+
+
+@given(
+    st.frozensets(st.text(max_size=200), max_size=100),
+    st.binary(max_size=500),
+    st.one_of(st.none(), st.booleans()),
+    st.booleans(),
+)
+def test_get_dummy_tx_payloads(
+    participants: FrozenSet[str],
+    value: bytes,
+    vote: Optional[bool],
+    is_value_none: bool,
+) -> None:
+    """Test `get_dummy_tx_payloads`."""
+    expected = [
+        DummyTxPayloadMatcher(
+            DummyTxPayload(
+                sender=agent,
+                value=(value or agent) if not is_value_none else value,
+                vote=vote,
+            )
+        )
+        for agent in sorted(participants)
+    ]
+
+    actual = get_dummy_tx_payloads(participants, value, vote, is_value_none)
+
+    assert len(actual) == len(expected) == len(participants)
+    assert all(
+        actual_payload == expected_payload
+        for actual_payload, expected_payload in zip(actual, expected)
+    )
+
+
+class TestDummyTxPayload:  # pylint: disable=too-few-public-methods
+    """Test class for `DummyTxPayload`"""
+
+    @staticmethod
+    @given(st.text(max_size=200), st.binary(max_size=500), st.booleans(), st.integers())
+    def test_properties(
+        sender: str, value: bytes, vote: bool, round_count: int
+    ) -> None:
+        """Test all the properties."""
+        dummy_tx_payload = DummyTxPayload(sender, value, vote, round_count)
+        assert dummy_tx_payload.value == value
+        assert dummy_tx_payload.vote == vote
+        assert dummy_tx_payload.data == {"value": value}
```

Comment:
yeah so you see the issue here? If we have a voting payload with a vote instead of value, it won't be in the data. Which it actually should be. If we use "value" instead of "vote" for voting payloads that would be resolved. It's not like "value" is in use already on voting payloads. 

Note that this vote attribute is also not compared in the `__eq__` operator of the BaseTxPayload referenced above.